include/seqan3/argument_parser/detail/format_parse.hpp

```diff
@@ -222,11 +222,30 @@ class format_parse : public format_base
         return (std::find_if(begin_it, end_of_options_it,
             [&] (const std::string & v)
             {
-                size_t id_size{(prepend_dash(id)).size()};
-                if (v.size() < id_size)
-                    return false; // cannot be the correct identifier
+                std::string full_id = prepend_dash(id);
 
-                return v.substr(0, id_size) == prepend_dash(id); // check if prefix of v is the same
+                if constexpr (std::same_as<id_type, char>) // short id
```

Comment:
Applies to both `prepend_dash` and the `if constexpr`

This also depends on how this function is used, but this won't work for short id's if the input is either a CStyle String or a std::string (Instead of `o` you pass `"o" or a `std::string`).

If this is intended, you could maybe add a sentence to the documentation that it is a strict requirement that the `id` must be passed as `char` to be identified as short id?

Reply:
Yes, it is a silent requirement, as I handled it that way all over the argument parser (the add option call takes char and std::string for them).
I think enforcing this somehow would require some refactoring, so will just document it for now.